workers/unity/Packages/com.improbable.gdk.mobile/Editor/MobileLauncherWindow.cs

```diff
@@ -0,0 +1,173 @@
+                    }
+                }
+
+                using (new EditorGUI.DisabledScope(simulatorNames.Length == 0))
+                {
+                    if (GUILayout.Button("Launch iOS app in Simulator"))
+                    {
+                        var availableSimulators = iOSLaunchUtils.RetrieveAvailableiOSSimulators();
+                        if (availableSimulators.TryGetValue(simulatorNames[simulatorNameIndex], out var simulatorUID))
+                        {
+                            iOSLaunchUtils.Launch(launchConfig.ShouldConnectLocally, simulatorUID, launchConfig.RuntimeIp, true);
+                        }
+                        else
+                        {
+                            simulatorNames = availableSimulators.Keys.ToArray();
+                            simulatorNameIndex = 0;
+                            Debug.LogError("Failed to launch app on selected simulator.");
+                        }
+                    }
+                }
+                
+                DrawHorizontalLine(8, DarkGrey);
+
+                using (new GUILayout.HorizontalScope())
+                {
+                    using (new EditorGUI.DisabledScope(deviceNames.Length == 0))
+                    {
+                        deviceNameIndex = EditorGUILayout.Popup("Device Model", deviceNameIndex, deviceNames);
+                    }
+
+                    var buttonIcon = new GUIContent(EditorGUIUtility.IconContent("Refresh"))
+                    {
+                        tooltip = "Refresh your device list."
+                    };
+
+                    if (GUILayout.Button(buttonIcon, EditorStyles.miniButton, GUILayout.ExpandWidth(false)))
+                    {
+                        deviceNames = iOSLaunchUtils.RetrieveAvailableiOSDevices().Keys.ToArray();
+                        deviceNameIndex = 0;
+                    }
+                }
+
+                using (new EditorGUI.DisabledScope(deviceNames.Length == 0))
+                {
+                    if (GUILayout.Button("Launch iOS app on device"))
+                    {
+                        var availableDevices = iOSLaunchUtils.RetrieveAvailableiOSDevices();
+                        if (availableDevices.TryGetValue(deviceNames[deviceNameIndex], out var deviceUID))
+                        {
+                            iOSLaunchUtils.Launch(launchConfig.ShouldConnectLocally, deviceUID, launchConfig.RuntimeIp, false);
+                        }
+                        else
+                        {
+                            deviceNames = availableDevices.Keys.ToArray();
+                            deviceNameIndex = 0;
+                            Debug.LogError("Failed to launch app on selected device. Is the device still connected?");
+                        }
+                    }
+                }
+            }
+#endif
+        }
+        
+        private void DrawHorizontalLine(int height, Color color)
```

Comment:
Don't we have this defined somewhere? I think we use it in multiple places

Reply:
it's slightly different from the one you used in the deployment launcher

Reply:
Could we merge them together?